src/array/ops/take.rs

```diff
@@ -19,10 +19,9 @@ where
             panic!("Out of bounds: {} vs len: {}", idx, self.len())
         }
         let arrow_array = self.downcast();
-        let is_valid = match arrow_array.validity() {
-            Some(validity) => validity.get_bit(idx),
-            None => true,
-        };
+        let is_valid = arrow_array
+            .validity()
+            .map_or(true, |validity| validity.get_bit(idx));
```

Comment:
Ah very cool! Didn't realize there was a `map_or` available, would clean up some of my own code 😛

Reply:
Yeah there area lot of nice `Result` and `Option` methods that provide a more declarative/functional API for extracting and/or transforming contained values that should make a good bit of our current imperative logic cleaner. I've definitely seen these APIs embraced across other projects, and it seems to lead to more concise and clear code!

https://doc.rust-lang.org/std/result/#method-overview